### ISSUE TITLE
Stream port ranges

### ISSUE THREAD

**Issue by pizzaandcheese** [NginxProxyManager/nginx-proxy-manager#1969]
have a simple addition I would like to request.
The ability to forward port ranges for streams.

According to the docs for Nginx it should be possible. https://nginx.org/en/docs/stream/ngx_stream_core_module.html#listen
Although when I attempt to forward a range in NPM it requires me to input 1 port number and won't accept a range.

**Comment by sanegaming** [comment]
I am also needing this feature, I have around 4,000 ports I need to setup and doing it one at a time is not really feasible. If anyone has any other work arounds please let me know.

**Comment by odjhl88** [comment]
So, I thought I'd have a look at the conf file generated by NPM. In my case I'm using the Alpine script for Proxmox. This puts the conf file in `/data/nginx/stream/x.conf` . Here's the code ouput for one of the files.
```
# ------------------------------------------------------------
# <PORT> TCP: 1 UDP: 1
# ------------------------------------------------------------


server {
  listen <PORT>;
listen [::]:<PORT>;


  proxy_pass <FORWARD_IP>:<PORT>;

  # Custom
  include /data/nginx/custom/server_stream[.]conf;
  include /data/nginx/custom/server_stream_tcp[.]conf;
}


server {
  listen <PORT> udp;
listen [::]:<PORT> udp;

  proxy_pass <FORWARD_IP>:<PORT>;

  # Custom
  include /data/nginx/custom/server_stream[.]conf;
  include /data/nginx/custom/server_stream_udp[.]conf;
}
```
So, I thought in my brilliance, I'd just modify the file with a port range at the end. Aparently, that's not how this works because now NPM doesn't load it's web interface at all. If someone thinks I might be on to something here, feel free to modify my screw up and see if this can be accomplished.
```
# ------------------------------------------------------------
# <PORT-START>-<PORT-END> TCP: 1 UDP: 1
# ------------------------------------------------------------


server {
  listen <PORT-START>-<PORT-END>;
listen [::]:<PORT-START>-<PORT-END>;


  proxy_pass <FORWARD_IP>:<PORT-START>-<PORT-END>;

  # Custom
  include /data/nginx/custom/server_stream[.]conf;
  include /data/nginx/custom/server_stream_tcp[.]conf;
}


server {
  listen <PORT-START>-<PORT-END> udp;
listen [::]:<PORT-START>-<PORT-END> udp;

  proxy_pass <FORWARD_IP>:<PORT-START>-<PORT-END>;

  # Custom
  include /data/nginx/custom/server_stream[.]conf;
  include /data/nginx/custom/server_stream_udp[.]conf;
}
```
Thanks in advance.

**Comment by sanegaming** [comment]
Please continue to consider.

**Comment by scroguard** [comment]
just had to enter a large range manually, this would be an amazing feature to have.

**Comment by MinisterPhobia** [comment]
I would like to see this added as well.

**Comment by Beanietech** [comment]
yes honestly please put this in. cause trying to port game servers is a huge pain..... and that is literally our last step before auto deployment servers....

**Comment by duzhen** [comment]
looking forward to have this functionalitiy

**Comment by scroguard** [comment]
this would be amazing to have, it's a real pain to have to enter each port of a large range individually.

**Comment by GamerClassN7** [comment]
Any news ?

**Comment by gr33k** [comment]
Trying to setup a stream to forward SIP ports to a PBX - I'd love to be able to add 10,000-20,000 UDP range...but as we all know - currently not possible. Please add this feature :)

**Comment by iamdagy** [comment]
@gr33k have you found a work around? im on the same boat trying to open many udp ports fora pbx.

**Comment by Jason733i** [comment]
Would also love this feature!

**Comment by jsapede** [comment]
would like this too

**Comment by paulmataruso** [comment]
Really would like this feature.

**Comment by aflorzy** [comment]
This would be a very convenient feature

**Comment by gr33k** [comment]
Thank you all for requesting this cool feature!!! Sadly - I haven't found a solution yet as the stream is a single port. 

Instead - I have two PBX...one with the main portforwards going to it. So I just setup a sip trunk between that PBX and the other one....Perfect? No....remote users can register a phone/softphone to the 2nd PBX - But it works good enough for testing. 

I suspect a SBC may be able to address this as well ... but I haven't had the time to explore this yet.

**Comment by pabloeltaco** [comment]
+1 for Port range forwarding. I have an FTP server and I need to forward 100 ports  manually :-(

**Comment by Rinpyre** [comment]
I'd also need this!

**Comment by gr33k** [comment]
Still hoping for port range in Streams ;)

**Comment by rikka-chunibyo** [comment]
PLEASE add this

**Comment by kaktuss77** [comment]
+1 please 

thanks

**Comment by aurelienizl** [comment]
Still waiting for this feature too!

**Comment by MrDevanWright** [comment]
Still waiting for this feature too!

**Comment by Keugon** [comment]
please

**Comment by ulyxie** [comment]
how is this not added yet :( I just wanted to setup traccar and I have to go around this limitation.

**Comment by gridphp** [comment]
Any maintainer alive?
